### PR TITLE
Server Access control layer filter

### PR DESCRIPTION
**Pull request by rldhont**
* **Funded by**: 3Liz
* **Description**: 
  * Implement `layerFilterExpression` based on `loginFilteredLayers`
  * Implement `layerFilterSubsetString` based on `loginFilteredLayers`

### CODE REVIEW

**Review by Gustry — APPROVED** [review]
Thanks !

**Comment by rldhont** [comment]
ready for review

File: test/server/test_lizmap_accesscontrol.py [hunk 1]
```diff
@@ -220,3 +219,139 @@ def test_group_visibility_headers(client):
...
+    rv = client.get(qs, projectfile, headers)
+    assert rv.status_code == 200
+    assert rv.headers.get('Content-Type', '').find('text/xml') == 0
+
+    layers = rv.xpath('//gml:featureMember')
+    assert layers is not None
+    assert len(layers) == 4
+
+    # Project with config with group filter
+    projectfile = "france_parts_liz_filter_group.qgs"
```
Comment (Gustry) [inline test/server/test_lizmap_accesscontrol.py:291]: Did you forget to commit this file?

Comment (rldhont) [reply test/server/test_lizmap_accesscontrol.py:291]: Yes!

Comment (rldhont) [reply test/server/test_lizmap_accesscontrol.py:291]: Done

**Review by Gustry — APPROVED** [review]
Nice work ! With tests provided, LGTM of course !

You added tests in both QGIS Desktop tests suite and QGIS Server tests suite. Just to be sure for you.
The `tools` are executed by QGIS Desktop test suites.

**Pull request merged**